### PR TITLE
Add DB, Routes, Docs, Schema

### PR DESCRIPTION
**Pull request by vladhrapov**
Add docs for mongo installation, project list schema, some routes for get and post project list data, configure app routes

### CODE REVIEW

**Comment by vladhrapov** [comment]
I used it for incoming data. I forgot delete.

File: server/routes/api/projects.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        }
+        console.log('There is some problems with getting data from DB!!!'.toUpperCase());
+    });
+
+});
+
+// This route for getting project data from client and saving them to DB
+router.post('/add', function(req, res){
+
+    var newProject = Project({
```
Comment (kucherenko) [inline server/routes/api/projects.js:20]: add verification of required fields and error response for not valid requests

File: server/routes/api/projects.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        name: req.body.name,
+        color: req.body.color,
+        taskCount: Math.floor((Math.random() * 10) + 1)    // This is for demo (random task count)
+    });
+
+    newProject.save(function(err){
+        if(err){
+            console.log('There is some problem with saving data to DB!!!'.toUpperCase());
```
Comment (kucherenko) [inline server/routes/api/projects.js:28]: if I'm a user I have no chance to see this message and if I'm developer and can see this message it is absolutely not interesting message :) make it more clear

File: server/routes/api/projects.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+});
+
+// This route for getting project data from client and saving them to DB
+router.post('/add', function(req, res){
+
+    var newProject = Project({
+        name: req.body.name,
+        color: req.body.color,
+        taskCount: Math.floor((Math.random() * 10) + 1)    // This is for demo (random task count)
```
Comment (kucherenko) [inline server/routes/api/projects.js:23]: move mocks to frontend part

**Pull request merged**